egs/wsj/s5/steps/nnet3/get_egs_targets.sh

```
@@ -349,15 +384,33 @@ if [ $stage -le 5 ]; then
   done
 
   if [ $archives_multiple == 1 ]; then # normal case.
+    if $generate_egs_scp; then
+      output_archive="ark,scp:$dir/egs.JOB.ark,$dir/egs.JOB.scp"
+    else
+      output_archive="ark:$dir/egs.JOB.ark"
+    fi
     $cmd --max-jobs-run $nj JOB=1:$num_archives_intermediate $dir/log/shuffle.JOB.log \
-      nnet3-shuffle-egs --srand=\$[JOB+$srand] "ark:cat $egs_list|" ark:$dir/egs.JOB.ark  || exit 1;
+      nnet3-shuffle-egs --srand=\$[JOB+$srand] "ark:cat $egs_list|" $output_archive  || exit 1;
+
+    if $generate_egs_scp; then
+      #concatenate egs.JOB.scp in single egs.scp
+      rm -rf $dir/egs.scp
```

remove -rf options, do '2>/dev/null || true' to suppress errors.